src/Data/List/Linear.hs

```diff
@@ -354,18 +354,34 @@ zip3 :: (Consumable a, Consumable b, Consumable c) => [a] %1 -> [b] %1 -> [c] %1
 zip3 = zipWith3 (,,)
 
 zipWith :: (Consumable a, Consumable b) => (a %1 -> b %1 -> c) -> [a] %1 -> [b] %1 -> [c]
-zipWith f xs ys =
-  zipWith' f xs ys & \(ret, leftovers) ->
-    leftovers `lseq` ret
+zipWith f =
+  zipWithk f (:) [] consume2 consume2
+  where
+    consume2 :: forall x y z. (Consumable x, Consumable y) => x %1 -> y %1 -> [z]
+    consume2 x y = x `lseq` y `lseq` []
```

Comment:
In `zipWith3` you do ```(x, y) `lseq` []``` instead, any reason to prefer one or the other?

Reply:
I don't think it makes much difference after optimisation. In principle, even without optimisation ``x `lseq` y `lseq` []`` is going to be one fewer allocation.